File: packages/webview_flutter/webview_flutter_android/android/src/test/java/android/net/Uri.java
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package android.net;

import androidx.annotation.NonNull;

// Creates an implementation of Uri that can be used with unittests and the JVM. Typically
// android.net.Uri does nothing when not used with an Android environment.
public class Uri {
  private final String url;

  public static Uri parse(String url) {
    return new Uri(url);
  }

  private Uri(String url) {
    this.url = url;
  }

  @NonNull
  @Override
  public String toString() {
    return url;
  }
}

File: packages/interactive_media_ads/android/src/test/kotlin/android/net/Uri.kt
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package android.net

/**
 * Redeclaration of Uri that works for tests.
 *
 * Without this redeclaration, `Uri.parse` always returns null.
 */
class Uri {
  companion object {
    @JvmStatic var lastValue: String? = null

    @JvmStatic
    fun parse(value: String): Uri {
      lastValue = value
      return Uri()
    }
  }
}
